gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/services/integrations/S3BucketIntegrationUpdateService.java

```diff
@@ -0,0 +1,144 @@
+                                            DocumentService documentService,
+                                            ProjectService projectService,
+                                            DocumentSectionService documentSectionService) {
+        this.sqsListener = sqsListener;
+        this.downloadService = downloadService;
+        this.documentService = documentService;
+        this.projectService = projectService;
+        this.documentSectionService = documentSectionService;
+    }
+
+    /**
+     * Checks for updates in the specified integration.
+     * Retrieves messages from SQS and processes them accordingly.
+     * @param integration The integration to check for updates.
+     * @return A list of multipart files representing the updated documents.
+     */
+    @Override
+    public List<MultipartFile> checkForUpdates(Integration integration) {
+
+        String queueName = integration.getQueueName();
+        List<MultipartFile> fileList = new ArrayList<>();
+
+        try {
+            List<Message> sqsMessages = sqsListener.receiveMessages(queueName);
+
+            for (Message sqsMessage : sqsMessages) {
+                handleSqsMessage(sqsMessage, fileList, integration);
+                sqsListener.deleteMessage(sqsMessage, queueName);
+            }
+        } catch (Exception e) {
+            logger.error("An error occurred while checking for updates: " + e.getMessage(), e);
+        }
+
+        return fileList;
+    }
+
+    /**
+     * Handles a single SQS message received from the integration queue.
+     * Parses the message and takes appropriate actions based on the event.
+     * @param sqsMessage The SQS message to handle.
+     * @param fileList The list to store updated multipart files.
+     * @param integration The integration associated with the message.
+     */
+    private void handleSqsMessage(Message sqsMessage, List<MultipartFile> fileList, Integration integration) {
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode rootNode = objectMapper.readTree(sqsMessage.getBody());
+
+            String event = rootNode.at(S3BucketIntegrationConstants.OBJECT_ROOT_EVENT).asText();
+            String bucketName = rootNode.at(S3BucketIntegrationConstants.OBJECT_ROOT_BUCKET_NAME).asText();
+            String objectKey = rootNode.at(S3BucketIntegrationConstants.OBJECT_ROOT_KEY).asText();
+
+            if (event.equals(S3BucketIntegrationConstants.EVENT_PUT_DOCUMENT)) {
+                logger.debug("Upload file from S3 bucket");
+                MultipartFile multipartFile = convertToMultipartFile(bucketName, objectKey);
+                fileList.add(multipartFile);
+            } else if (event.equals(S3BucketIntegrationConstants.EVENT_DELETE_DOCUMENT)) {
+                logger.debug("Deleted file from S3 bucket");
+                handleDeletedDocuments(integration, objectKey);
+            }
+        } catch (IOException | GendoxException e) {
```

Comment:
This exception is swallowed, It is not thrown. So, even if there is an error here, the message will be deleted in the parent method.